src/main/java/uk/ac/ic/wlgitbridge/git/util/RepositoryObjectTreeWalker.java

```diff
@@ -46,7 +49,14 @@ public RepositoryObjectTreeWalker(
 
     public RawDirectory getDirectoryContents(
     ) throws IOException, SizeLimitExceededException, InvalidGitRepository {
-        return new RawDirectory(walkGitObjectTree());
+        return getDirectoryContents(50 * 1024 * 1024);
```

Comment:
I am not sure whether 50MB will be the "right" size or not. Could we extract this into a config var?

Reply:
Oops. I had not read ahead to part 4. You can ignore this one.